pkg/ruler/manager.go

```diff
@@ -184,6 +198,29 @@ func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user
 	}
 }
 
+func (r *DefaultMultiTenantManager) getRulesManager(user string, ctx context.Context) (RulesManager, bool) {
```

Comment:
Can we only do one thing in this method to get manager from the map rather than creating a new one?
Create manager and make it up and running should be a separate function.